packages/medusa/src/api/routes/store/carts/create-cart.ts

```diff
@@ -81,26 +81,24 @@ export default async (req, res) => {
   const entityManager: EntityManager = req.scope.resolve("manager")
   const featureFlagRouter: FlagRouter = req.scope.resolve("featureFlagRouter")
 
-  await entityManager.transaction(async (manager) => {
-    let regionId: string
-    if (typeof validated.region_id !== "undefined") {
-      regionId = validated.region_id
-    } else {
-      const regions = await regionService
-        .withTransaction(manager)
-        .list({})
-
-      if (!regions?.length) {
-        throw new MedusaError(
-          MedusaError.Types.INVALID_DATA,
-          `A region is required to create a cart`
-        )
-      }
-
-      regionId = regions[0].id
+  let regionId: string
+  if (typeof validated.region_id !== "undefined") {
+    regionId = validated.region_id
+  } else {
+    const regions = await regionService.list({})
+
+    if (!regions?.length) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `A region is required to create a cart`
+      )
     }
 
-    let cart = await cartService.withTransaction(manager).create({
+    regionId = regions[0].id
+  }
```

Comment:
thought(non-blocking): this is a good candidate for a validator middleware; could be nice with a separate PR for this.

Reply:
yes indeed, but as you proposed, a separate pr would be a better fit I think. Should we create an issue for that?